src/linear_algebra/pi_tensor_product.lean

```diff
@@ -354,9 +357,80 @@ theorem lift.unique {φ' : (⨂[R] i, s i) →ₗ[R] E} (H : ∀ f, φ' (tprod R
   φ' = lift φ :=
 lift.unique' (multilinear_map.ext H)
 
+@[simp]
+theorem lift_symm (φ' : (⨂[R] i, s i) →ₗ[R] E) : lift.symm φ' = φ'.comp_multilinear_map (tprod R) :=
+rfl
+
+@[simp]
 theorem lift_tprod : lift (tprod R : multilinear_map R s _) = linear_map.id :=
 eq.symm $ lift.unique' rfl
 
+section
+variables (R M)
+/-- Re-index the components of the tensor power by `e`.
+
+For simplicity, this is defined only for homegenously- (rather than dependently-) typed components.
+-/
+def reindex (e : ι ≃ ι₂) : ⨂[R] i : ι, M ≃ₗ[R] ⨂[R] i : ι₂, M :=
+linear_equiv.of_linear
+  ((lift.symm.trans $
+    multilinear_map.dom_dom_congr_linear_equiv M (⨂[R] i : ι₂, M) R R e.symm).trans
+      lift (linear_map.id))
+  ((lift.symm.trans $
+    multilinear_map.dom_dom_congr_linear_equiv M (⨂[R] i : ι, M) R R e).trans
+      lift (linear_map.id))
+  (by { ext, simp })
+  (by { ext, simp })
+end
+
+@[simp] lemma reindex_tprod (e : ι ≃ ι₂) (f : Π i, M) :
+  reindex R M e (tprod R f) = tprod R (λ i, f (e.symm i)) :=
+lift.tprod f
+
+@[simp] lemma reindex_comp_tprod (e : ι ≃ ι₂) :
+  (reindex R M e : ⨂[R] i : ι, M →ₗ[R] ⨂[R] i : ι₂, M).comp_multilinear_map (tprod R) =
+    (tprod R : multilinear_map R (λ i, M) _).dom_dom_congr e.symm :=
+multilinear_map.ext $ reindex_tprod e
+
+@[simp] lemma lift_comp_reindex (e : ι ≃ ι₂) (φ : multilinear_map R (λ _ : ι₂, M) E) :
+  (lift φ).comp ↑(reindex R M e) = lift (φ.dom_dom_congr e.symm) :=
+by { ext, simp, }
+
+@[simp]
+lemma lift_reindex (e : ι ≃ ι₂) (φ : multilinear_map R (λ _, M) E) (x : ⨂[R] i, M) :
+  lift φ (reindex R M e x) = lift (φ.dom_dom_congr e.symm) x :=
+linear_map.congr_fun (lift_comp_reindex e φ) x
+
+@[simp] lemma reindex_symm (e : ι ≃ ι₂) :
+  reindex R M e.symm = (reindex R M e).symm := rfl
```

Comment:
Is it clear that this should be a simp lemma? I'm not even sure which way I would want to use this one most often actually -- and the next simp lemma seems to go the other way.

Reply:
I was intending to model the simp lemmas after [`equiv.perm.sigma_congr_right_(trans|symm|refl)`](https://leanprover-community.github.io/mathlib_docs/data/equiv/basic.html#equiv.perm.sigma_congr_right_trans). So yes, this one is backwards - the idea is to push the `equiv` operations to the same place in the hope that they cancel out.

Reply:
I see -- looks good now!